### PR TITLE
formula_cellar_checks: show mismatched arch in `check_binary_arches`

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

This will make the error more informative by showing the architecture a
binary was built for along with the error message.

Before:

    foo:
      * Binaries built for an incompatible architecture were installed into foo's prefix.
        The offending files are:
          /usr/local/Cellar/foo/1.0/lib/libbar.dylib
          /usr/local/Cellar/foo/1.0/lib/libfoo.dylib
          /usr/local/Cellar/foo/1.0/lib/libincompatible.dylib
        Unexpected universal binaries were found.
        The offending files are:
          /usr/local/Cellar/foo/1.0/lib/liball.dylib
          /usr/local/Cellar/foo/1.0/lib/libuniversal.dylib

After:

    foo:
      * Binaries built for a non-native architecture were installed into foo's prefix.
        The offending files are:
          /usr/local/Cellar/foo/1.0/lib/libbar.dylib        (i386)
          /usr/local/Cellar/foo/1.0/lib/libfoo.dylib        (arm64)
          /usr/local/Cellar/foo/1.0/lib/libincompatible.dylib       (universal)
        Unexpected universal binaries were found.
        The offending files are:
          /usr/local/Cellar/foo/1.0/lib/liball.dylib
          /usr/local/Cellar/foo/1.0/lib/libuniversal.dylib

### CODE REVIEW

File: Library/Homebrew/formula_cellar_checks.rb [hunk 1]
```diff
@@ -340,17 +342,17 @@ def check_binary_arches(formula)
 
     if mismatches.present?
       s += <<~EOS
-        Binaries built for an incompatible architecture were installed into #{formula}'s prefix.
+        Binaries built for a non-native architecture were installed into #{formula}'s prefix.
         The offending files are:
-          #{mismatches * "\n  "}
+          #{mismatches.map { |m| "#{m.first}\t(#{m.last})" } * "\n  "}
```
Comment (carlocab) [inline Library/Homebrew/formula_cellar_checks.rb:347]: The `\t` is a crude attempt at getting some sort of alignment of the output. It's not perfect, but I think it suits for what we're using this for.

Also, this still shows `(universal)` as the arch for universal binaries that don't contain a native slice. Not sure if we want to show the actual slices, so I've left it this way for now.

Comment (Rylan12) [reply Library/Homebrew/formula_cellar_checks.rb:347]: I think this is fine.

Another option could be to put the type first:

```
foo:
  * Binaries built for a non-native architecture were installed into foo's prefix.
    The offending files are:
      (i386)          /usr/local/Cellar/foo/1.0/lib/libbar.dylib
      (arm64)         /usr/local/Cellar/foo/1.0/lib/libfoo.dylib
      (universal)     /usr/local/Cellar/foo/1.0/lib/libincompatible.dylib
    Unexpected universal binaries were found.
    The offending files are:
      /usr/local/Cellar/foo/1.0/lib/liball.dylib
      /usr/local/Cellar/foo/1.0/lib/libuniversal.dylib
```

I'm not sure I like this any better, though. Probably just a personal preference thing

**Review by Rylan12 — APPROVED** [review]

File: Library/Homebrew/formula_cellar_checks.rb (hunk 1 above)
Comment (carlocab) [reply Library/Homebrew/formula_cellar_checks.rb:347]: Yea, I'm fine with slightly misaligned output when some `Pathname`s are too long. It's still readable anyway, which is what I was going for with alignment (compare this with the case where I only leave a single space as in `"#{m.first} (#{m.last})"`).

**Comment by MikeMcQuaid** [comment]
Nice work @carlocab!

**Pull request merged**